src/jdk.crypto.ec/share/classes/sun/security/ec/SunEC.java

```diff
@@ -161,7 +174,24 @@ public Object newInstance(Object ctrParamObj)
                     }
                 } else  if (type.equals("KeyPairGenerator")) {
                     if (algo.equals("EC")) {
-                        return new ECKeyPairGenerator();
+                        /* Disabling OpenSSL usage in AIX due to perfomance regression observed */
+                        if (useNativeECKeyGen &&
+                           (NativeCrypto.getVersionIfAvailable() >= 0x10100000) &&
+                           (!System.getProperty("os.name").contains("AIX"))) {
+                            if (nativeCryptTrace) {
+                                System.err.println("EC KeyPair Generation - Using OpenSSL integration.");
+                            }
+                            return new NativeECKeyPairGenerator();
+                        } else {
+                            if (nativeCryptTrace) {
+                                if (System.getProperty("os.name").contains("AIX")) {
+                                    System.err.println("EC KeyPair Generation - Not using OpenSSL integration in AIX.");
```

Comment:
What's special about AIX?

Reply:
There was a regression for that functionality in AIX, even with higher OpenSSL versions, so we decided to go with the Java implementation to avoid that.

Reply:
Please include a comment explaining that.